### PR TITLE
CMR-6096

### PR DESCRIPTION
**Pull request by ygliuvt**
This is the PR to merge the update es cluster feature branch.
It is a collection of PRs that have already been approved for the ES upgrade task.

### CODE REVIEW

**Comment by mschmele** [comment]
I would like to go over this PR via a goto just because of how big it is if for no reason other than to get a relatively quick overview of it

**Comment by ygliuvt** [comment]
It is a collection of PRs that have already been submitted and approved for the ES upgrade task.
> I would like to go over this PR via a goto just because of how big it is if for no reason other than to get a relatively quick overview of it

File: search-app/src/cmr/search/services/query_service.clj [hunk 1]
```diff
@@ -275,7 +275,7 @@
                                                       :concept-id concept-id
                                                       :result-format result-format})
           results (qe/execute-query context query)]
-      (when (zero? (:hits results))
+      (when (zero? (get-in results [:hits]))
```
Comment (indiejames) [inline search-app/src/cmr/search/services/query_service.clj:278]: How is this different from `(when (zero? (:hits results))`?

File: search-app/src/cmr/search/services/query_service.clj [hunk 2]
```diff
@@ -386,7 +386,7 @@
                                               (assoc :all-revisions? true)
                                               (assoc :condition condition)
                                               (assoc :page-size :unlimited)))
-        coll-concept-ids (map #(first (:concept-id (:fields %)))
+        coll-concept-ids (map #(:concept-id (:_source %))
```
Comment (indiejames) [inline search-app/src/cmr/search/services/query_service.clj:389]: I think `get-in % [:_source :concept-id]` would be slightly cleaner.

File: search-app/src/cmr/search/services/query_service.clj (hunk 1 above)
Comment (ygliuvt) [reply search-app/src/cmr/search/services/query_service.clj:278]: I don't know. I guess it is a result of multiple commits. At some point it was `(when (zero? (get-in results [:hits :hits])`. It shouldn't have changed.

**Pull request merged**